src/Nowruz/modules/general/components/fileUploaderMultiple/useFileUploaderMultiple.tsx

```diff
@@ -60,33 +63,54 @@ export const useFileUploader = (
     if (fileType === 'application/pdf') return pdf;
     if (fileType === 'image/jpeg') return jpg;
   };
-  const onDrop = (acceptedFiles: File[]) => {
-    const newUploadedSize = acceptedFiles.map(f => f.size).reduce((a, b) => a + b, 0);
-    const newTotal = totalSize + newUploadedSize;
-    if (newTotal > maxSize * 1024 * 1024) {
-      setError(`Max file size is ${maxSize}mb`);
-      return;
-    }
+  const onDrop = async (acceptedFiles: File[]) => {
+    try {
+      setProgress(0);
+      setError('');
+      const requests: Promise<PostMediaUploadRes>[] = [];
+      acceptedFiles.forEach(f => {
+        if (f.size > maxSize * 1024 * 1024) {
+          setError(`Max file size is ${maxSize}mb`);
+          return;
+        } else {
+          requests.push(uploadMediaWithProgress(f, setProgress));
+        }
+      });
 
-    setTotalSize(newTotal);
-    setFiles([...files, ...acceptedFiles]);
+      setUploading(true);
+      const res = await Promise.all(requests);
+      setUploaded([...uploaded, ...res]);
+      setFiles([...files, ...acceptedFiles]);
+      setUploading(false);
```

Comment:
`setUploading` called in both try and catch clause, please move it out side of try/catch.